parquet-column/src/main/java/org/apache/parquet/column/statistics/SizeStatistics.java

```diff
@@ -0,0 +1,214 @@
+     * It is called when value is null, or the column is not of BYTE_ARRAY type.
+     *
+     * @param repetitionLevel repetition level of the value
+     * @param definitionLevel definition level of the value
+     */
+    public void add(int repetitionLevel, int definitionLevel) {
+      assert repetitionLevel < repetitionLevelHistogram.length;
+      assert definitionLevel < definitionLevelHistogram.length;
+      repetitionLevelHistogram[repetitionLevel]++;
+      definitionLevelHistogram[definitionLevel]++;
+    }
+
+    /**
+     * Add repetition and definition level of a value to the statistics.
+     * It is called when value is null, or the column is not of BYTE_ARRAY type.
+     *
+     * @param repetitionLevel repetition level of the value
+     * @param definitionLevel definition level of the value
+     * @param value value of to be added
+     */
+    public void add(int repetitionLevel, int definitionLevel, Binary value) {
+      add(repetitionLevel, definitionLevel);
+      if (type.getPrimitiveTypeName() == PrimitiveType.PrimitiveTypeName.BINARY) {
+        unencodedByteArrayDataBytes = Math.addExact(unencodedByteArrayDataBytes, value.length());
+      }
+    }
+
+    /**
+     * Build a SizeStatistics from the builder.
+     */
+    public SizeStatistics build() {
+      return new SizeStatistics(type, unencodedByteArrayDataBytes,
+        new LongArrayList(repetitionLevelHistogram), new LongArrayList(definitionLevelHistogram));
+    }
+
+  }
+
+  /**
+   * Create a SizeStatistics.
+   *
+   * @param type physical type of the column associated with this statistics
+   * @param unencodedByteArrayDataBytes number of physical bytes stored for BYTE_ARRAY data values assuming no encoding
+   * @param repetitionLevelHistogram histogram for all repetition levels if non-empty
+   * @param definitionLevelHistogram histogram for all definition levels if non-empty
+   */
+  public SizeStatistics(PrimitiveType type,
+                        long unencodedByteArrayDataBytes,
+                        List<Long> repetitionLevelHistogram,
+                        List<Long> definitionLevelHistogram) {
+    this.type = type;
+    this.unencodedByteArrayDataBytes = unencodedByteArrayDataBytes;
+    this.repetitionLevelHistogram = repetitionLevelHistogram;
+    this.definitionLevelHistogram = definitionLevelHistogram;
+  }
+
+  /**
+   * Merge two SizeStatistics of the same column.
+   * It is used to merge size statistics from all pages of the same column chunk.
+   */
+  public void mergeStatistics(SizeStatistics other) {
+    Preconditions.checkArgument(type.equals(other.type), "Cannot merge SizeStatistics of different types");
+    Preconditions.checkArgument(repetitionLevelHistogram.size() != other.repetitionLevelHistogram.size(),
+      "Cannot merge repetitionLevelHistogram with different sizes");
+    Preconditions.checkArgument(definitionLevelHistogram.size() != other.definitionLevelHistogram.size(),
```

Comment:
```suggestion
    Preconditions.checkArgument(repetitionLevelHistogram.size() == other.repetitionLevelHistogram.size(),
      "Cannot merge repetitionLevelHistogram with different sizes");
    Preconditions.checkArgument(definitionLevelHistogram.size() == other.definitionLevelHistogram.size(),
```
This one was hard to track down :sweat_smile:

Reply:
Thanks for pointing it out! I will add more teats later.